LibreNMS/Config.php

```diff
@@ -137,6 +138,67 @@ public static function get($key, $default = null)
         return Arr::get(self::$config, $key, $default);
     }
 
+    public static function getExecutable(string $key): string
+    {
+        $binary = self::get($key, $key);
+
+        if (Str::startsWith($binary, '/')) {
+            return $binary;
+        }
+
+        $process = new Process(['whereis', '-b', $binary]);
+        $process->run();
+        if ($process->getExitCode() === 0) {
+            $output = trim($process->getOutput());
+            $list = substr($output, strpos($output, ':') + 1);
+            $possible_targets = explode(' ', $list);
+        } else {
+            $process = new Process(['which', $binary]);
+            $process->run();
+            $possible_targets = [trim($process->getOutput())];
+        }
+
+        foreach ($possible_targets as $target) {
+            if (is_executable($target)) {
+                self::persist($key, $target);
+
+                return $target;
+            }
+        }
+
+        if (Eloquent::isConnected()) {
+            $notification = new Notification();
```

Comment:
Port this to the new \LibreNMS\Util\Notifications helper?

It would be nice to remove the notification if the issue is resolved.

Reply:
Sure but maybe for next release then